zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/distribution/CommandDistributionBehavior.java

```diff
@@ -149,7 +160,39 @@ public <T extends UnifiedRecordValue> void distributeCommand(
         (partition) -> distributeToPartition(partition, distributionRecord, distributionKey));
   }
 
-  private <T extends UnifiedRecordValue> void distributeToPartition(
+  private void distributeToPartition(
+      final int partition,
+      final CommandDistributionRecord distributionRecord,
+      final long distributionKey) {
+    final var distributionQueue = Optional.ofNullable(distributionRecord.getQueueId());
+    distributionQueue.ifPresent(queue -> enqueueDistribution(queue, partition, distributionKey));
+
+    final var otherQueuedDistributions =
+        distributionQueue
+            .flatMap(queue -> distributionState.nextQueuedDistributionKey(queue, partition))
+            .filter(nextDistributionKey -> nextDistributionKey != distributionKey);
+
+    if (otherQueuedDistributions.isEmpty()) {
```

Comment:
🙃 I'd prefer a comment here mentioning why we only distribute when it's empty. Since I'm familiar with the topic I know this is because there's a different distribution in the same queue that hasn't been acknowledged yet. But I can imagine this confuses someone who reads this for the first time.